runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -3804,6 +3805,19 @@ void jitHookClassLoadHelper(J9VMThread *vmThread,
       deserializer->onClassLoad(cl, vmThread);
 #endif /* defined(J9VM_OPT_JITSERVER) */
 
+   if (auto dependencyTable = compInfo->getPersistentInfo()->getAOTDependencyTable())
+      {
+      getClassNameIfNecessary(vm, clazz, className, classNameLen);
+      // These two classes and java/lang/J9VMInternals$ClassInitializationLock are
+      // the only ones that are marked initialized without going through the normal
+      // hooks (see initializeRequiredClasses() in jclcinit.c), and that last class
+      // is never stored in the SCC. For these two, this is both a load and an
+      // initialization.
+      bool isClassInitialization = (classNameLen == 17 && !memcmp(className, "com/ibm/oti/vm/VM", classNameLen)) ||
+                                   (classNameLen == 23 && !memcmp(className, "java/lang/J9VMInternals", classNameLen));
+      compInfo->getPersistentInfo()->getAOTDependencyTable()->classLoadEvent(clazz, true, isClassInitialization);
```

Comment:
Shouldn't you use `dependencyTable` here since you've already stored it to a local?